### PR TITLE
feat: Name on registration / Name scraping

### PR DESCRIPTION
**Pull request by force10269**
Since we added the `:name` field for our users and we want to have the user's name available to us for when we display orders in a shared cart later on, we added a name field for both registration and the settings page. On Google registration, we should now be grabbing a name from the Google callback. Here are screenshots of the changes made:

Screen Shot 2022-07-18 at 11 32 24 AM

Screen Shot 2022-07-18 at 11 32 29 AM

Screen Shot 2022-07-18 at 11 32 34 AM

### CODE REVIEW

File: lib/lunchbot_web/helpers/name_helper.ex [hunk 1]
```diff
@@ -0,0 +1,10 @@
+defmodule LunchbotWeb.NameHelper do
+  def getName(email) do
+    Regex.scan(~r/^[^@]*/, email)
+    |> Enum.at(0)
+    |> Enum.at(0)
+    |> upcaseFirst()
+  end
```
Comment (chewnoill) [inline lib/lunchbot_web/helpers/name_helper.ex]: Do we need this if we can get the name from the profile?

File: lib/lunchbot_web/controllers/user_registration_controller.ex [hunk 2]
```diff
@@ -5,12 +5,26 @@ defmodule LunchbotWeb.UserRegistrationController do
...
 
   def create(conn, %{"user" => user_params}) do
+    # If the user does not enter their name, scrape from email
+    user_params =
+      if(Map.get(user_params, "name") == "") do
+        userName =
+          Map.get(user_params, "email")
+          |> getName()
+
+        Map.replace(user_params, "name", userName)
```
Comment (chewnoill) [inline lib/lunchbot_web/controllers/user_registration_controller.ex]: Do we need to store the scrapped name in the database?

File: lib/lunchbot_web/templates/user_registration/new.html.heex [hunk 3]
```diff
@@ -7,6 +7,10 @@
     </div>
   <% end %>
 
+  <%= label f, "Full Name" %>
```
Comment (cathyzoller) [inline lib/lunchbot_web/templates/user_registration/new.html.heex]: Could this be simply "Your name"?

File: lib/lunchbot/accounts.ex [hunk 4]
```diff
@@ -210,6 +210,25 @@ defmodule Lunchbot.Accounts do
     end
   end
 
+  def change_user_name(user, attrs \\ %{}) do
+    User.name_changeset(user, attrs)
+  end
+
+  def update_user_name(user, attrs) do
```
Comment (cathyzoller) [inline lib/lunchbot/accounts.ex:217]: nice!

**Review by RVRX — APPROVED** [review]
LGTM

**Review by cathyzoller — APPROVED** [review]
Needs a rebase, otherwise looks good!

**Review by chewnoill — APPROVED** [review]
🚀

**Pull request merged**